klog.go

```diff
@@ -1416,6 +1416,18 @@ func Warningf(format string, args ...interface{}) {
 	logging.printf(warningLog, logging.logr, format, args...)
 }
 
+// WarningS structured logs to the INFO and WARNING logs.
+// The msg argument used to add constant description to the log line.
+// The key/value pairs would be join by "=" ; a newline is always appended.
+//
+// Basic examples:
+// >> klog.WarningS("Page size is invalid", "page_size", "something", "error", err)
+// output:
+// >> W1025 00:15:15.525108       1 controller_utils.go:351] "Page size is invalid" page_size="something" error="test error"
+func WarningS(msg string, keysAndValues ...interface{}) {
+	logging.warningS(msg, keysAndValues...)
```

Comment:
This change is incompatible with https://github.com/go-logr/logr interface and would not work with `SetLogger` interface.